app/controllers/shopify_app/callback_controller.rb

```diff
@@ -6,6 +6,11 @@ class CallbackController < ActionController::Base
     include ShopifyApp::LoginProtection
 
     def callback
+      if jwt_shopify_domain || jwt_shopify_user_id
+        jwt_callback
+        return
+      end
+
       if auth_hash
         login_shop
 
```

Comment:
Should this block of code be extracted out for single-level-of-abstraction reasons? It looks like we're doing that with `jwt_callback` above, and it seems nice.

Reply:
I tried cleaning this up for clarity, take a look and see what you think!